File: library_greendao/src/main/java/com/phone/library_greendao/bean/UserBean.java
package com.phone.library_greendao.bean;

import com.phone.library_greendao.AddressBeanListConverter;

import org.greenrobot.greendao.annotation.Convert;
import org.greenrobot.greendao.annotation.Entity;
import org.greenrobot.greendao.annotation.Generated;
import org.greenrobot.greendao.annotation.Id;

import java.util.ArrayList;
import java.util.List;

@Entity
public class UserBean implements Cloneable {

    @Id(autoincrement = true)
    private Long id;
    private String userId;
    private String password;
    private String birthday;
    private Integer age;
    private Double salary;
    @Convert(columnType = String.class, converter = AddressBeanListConverter.class)
    private List<AddressBean> addressBeanList = new ArrayList<>();

    @Generated(hash = 703739529)
    public UserBean(Long id, String userId, String password, String birthday,
                    Integer age, Double salary, List<AddressBean> addressBeanList) {
        this.id = id;
        this.userId = userId;
        this.password = password;
        this.birthday = birthday;
        this.age = age;
        this.salary = salary;
        this.addressBeanList = addressBeanList;
    }

    @Generated(hash = 1203313951)
    public UserBean() {
    }

    public Long getId() {
        return this.id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getUserId() {
        return this.userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getPassword() {
        return this.password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getBirthday() {
        return this.birthday;
    }

    public void setBirthday(String birthday) {
        this.birthday = birthday;
    }

    public Integer getAge() {
        return this.age;
    }

    public void setAge(Integer age) {
        this.age = age;
    }

    public Double getSalary() {
        return this.salary;
    }

    public void setSalary(Double salary) {
        this.salary = salary;
    }

    public List<AddressBean> getAddressBeanList() {
        return this.addressBeanList;
    }

    public void setAddressBeanList(List<AddressBean> addressBeanList) {
        this.addressBeanList = addressBeanList;
    }

    @Override
    protected UserBean clone() throws CloneNotSupportedException {
        List<AddressBean> addressBeanList = new ArrayList<>();
        UserBean userBean = (UserBean) super.clone();
        for (int i = 0; i < userBean.getAddressBeanList().size(); i++) {
            addressBeanList.add(userBean.getAddressBeanList().get(i).clone());
        }
        userBean.setAddressBeanList(addressBeanList);
        return userBean;
    }

}

File: library_common/src/main/java/com/phone/library_common/bean/UserBean.kt
package com.phone.library_common.bean

import org.greenrobot.greendao.annotation.Convert
import org.greenrobot.greendao.annotation.Entity
import org.greenrobot.greendao.annotation.Generated
import org.greenrobot.greendao.annotation.Id

@Entity
open class UserBean : Cloneable {

    @Id(autoincrement = true)
    var id: Long? = null
    var userId: String? = null
    var password: String? = null
    var birthday: String? = null
    var age: Int? = null
    var salary: Double? = null

    @Convert(columnType = String::class, converter = AddressBeanListConverter::class)
    var addressBeanList: List<AddressBean> = ArrayList()

    @Generated(hash = 703739529)
    constructor(
        id: Long?, userId: String?, password: String?, birthday: String?,
        age: Int?, salary: Double?, addressBeanList: List<AddressBean>
    ) {
        this.id = id
        this.userId = userId
        this.password = password
        this.birthday = birthday
        this.age = age
        this.salary = salary
        this.addressBeanList = addressBeanList
    }

    @Generated(hash = 1203313951)
    constructor() {
    }

    @Throws(CloneNotSupportedException::class)
    override fun clone(): UserBean {
        val addressBeanList: MutableList<AddressBean> = ArrayList()
        val userBean = super.clone() as UserBean
        for (i in userBean.addressBeanList.indices) {
            addressBeanList.add(userBean.addressBeanList[i].clone())
        }
        userBean.addressBeanList = addressBeanList
        return userBean
    }

}